### PR TITLE
Issue 51 sync with backend main experts view

### PR DESCRIPTION
**Pull request by DelFlumen**
develop

## GitHub Board

**Issue link**

[#51 ]( )

**Story link**

## Code reviewers
- [ ] @github_username

## Summary of change

- Synced Main ExpertsView with backend
- Edited ExpertDataCard
- Added loading spinner for Main Experts View
- Added error message for Main Experts View

### CODE REVIEW

File: src/modules/main/store/mainSlice.ts [hunk 1]
```diff
@@ -125,6 +154,20 @@ export const mainSlice = createSlice({
 
       state.newest.meta.loading = LoadingStatusEnum.failed;
     });
+    builder.addCase(fetchExperts.pending, (state) => {
+      state.experts.meta.loading = LoadingStatusEnum.pending;
+    });
+    builder.addCase(fetchExperts.fulfilled, (state, { payload }) => {
+      state.experts.meta.loading = LoadingStatusEnum.succeeded;
+      state.experts.experts.push(...payload);
```
Comment (Gtes) [inline src/modules/main/store/mainSlice.ts:162]:
```suggestion
      state.experts.experts = payload;
```

Comment (DelFlumen) [reply src/modules/main/store/mainSlice.ts:162]: fixed

File: src/modules/main/components/MainView.tsx [hunk 2]
```diff
@@ -1,20 +1,34 @@
...
+  const {
+    experts,
+    meta: { loading },
+  } = useSelector(selectMainExperts);
+
+  const renderError = (errorMsg = 'Не вдалося завантажити експертів') =>
+    loading === LoadingStatusEnum.failed ? <span>{errorMsg}</span> : null;
+
+  const renderLoading = () =>
+    loading === LoadingStatusEnum.pending ? <CircularProgress /> : null;
```
[the remark this answers is not in the document]
Comment (DelFlumen) [reply src/modules/main/components/MainView.tsx:31]: I'll reuse it after you merge your branch with develop.

File: src/modules/main/components/MainView.tsx [hunk 3]
```diff
@@ -27,7 +41,12 @@ const MainView: React.FC<IMainViewProps> = () => {
             <NewestContainer />
           </Grid>
           <Grid item xs={12}>
-            <ExpertsView cards={expertsCards} />
+            <Grid container direction="column" alignItems="center">
+              {renderLoading()}
+              {renderError()}
+            </Grid>
```
[the remark this answers is not in the document]
Comment (DelFlumen) [reply src/modules/main/components/MainView.tsx:47]: fixed

**Review by SergeOlabin — APPROVED** [review]

**Pull request merged**